Source/Core/Core/Core.cpp

```diff
@@ -707,11 +707,18 @@ void SetState(Core::System& system, State state, bool report_state_change)
   switch (state)
   {
   case State::Paused:
+#ifdef USE_RETRO_ACHIEVEMENTS
+    if (!AchievementManager::GetInstance().CanPause())
+      return;
```

Comment:
You should skip this check when called from CPUSetInitialExecutionState. Otherwise the state might stay as Starting when it shouldn't.